src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -130,18 +135,30 @@ export class KernelProcess implements IKernelProcess {
                 }
                 traceError('Kernel died', error, stderr);
                 if (error instanceof PythonKernelDiedError) {
+                    providedExitCode = error.exitCode;
```

Comment:
This `error` was getting hit, but we didn't get `PythonKernelDiedError` as this callback is only invoked once.
The PR (change) now ensures this will get called only once with `PythonKernelDiedError`